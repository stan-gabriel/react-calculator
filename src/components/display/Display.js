import React from 'react';
import './Display.scss';

class Display extends React.Component {

	adjustText = function (textLength) {
		if (textLength < 8) {
			return 3
		} else {
			return 24 / textLength
		}
	};

	render() {
		let initialLength = `${this.props.result} `;
		return (
			<div className="Display" style={{fontSize: `${this.adjustText(initialLength.toString().length)}rem`}}>
				{this.props.result}
			</div>
		)
	};
}

export default Display
