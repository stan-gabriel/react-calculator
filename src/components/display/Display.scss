@import '../../scss/variables';

.Display {
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  background: $base-color;
  height: 5rem;
  width: 14.4rem;
  font-weight: 200;
  color: #ffff;
  //font-size: 2rem;
  font-size: 3.5vh;
  padding: 0 0.4rem;
  border-top-right-radius: 5px;
  border-top-left-radius: 5px;

  white-space: pre-line;
  word-wrap: break-word;
}
