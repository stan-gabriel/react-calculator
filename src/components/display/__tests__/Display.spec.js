import React from 'react';
import { shallow } from 'enzyme';
import Display from '../Display';

describe('Display component', () => {
	let wrapper;
	beforeEach(() => (wrapper = shallow(<Display />)));
	test('should match the snapshot', () => expect(wrapper).toMatchSnapshot());

	test('adjustText function check', () => {
		const spy = jest.spyOn(wrapper.instance(), 'adjustText');
		expect(spy).toHaveBeenCalledTimes(0);
		wrapper.instance().adjustText();
		expect(spy).toHaveBeenCalledTimes(1);
		expect(wrapper.instance().adjustText(1)).toEqual(3)
	});

	test('should render a <div />', () => {
		expect(wrapper.find('div').length).toEqual(1);
	});

	test('should render the container', () => {
		expect(wrapper.find('.Display').length).toEqual(1);
	});
});
