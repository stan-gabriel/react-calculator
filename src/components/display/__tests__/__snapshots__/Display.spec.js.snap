// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component should match the snapshot 1`] = `
<div
  className="Display"
  style={
    Object {
      "fontSize": "2.4rem",
    }
  }
/>
`;
